addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3434,8 +3434,10 @@ registry.ImageOptimize = ImageHandlerOption.extend({
      * @override
      */
     _computeMaxDisplayWidth() {
+        // (naturalWidth / 3) is the minimum width that we can suggest to avoid bad rendering
         // TODO: read widths from computed style in case container widths are not default
-        const displayWidth = this._getImg().clientWidth;
+        const img = this._getImg();
+        const displayWidth = Math.max(img.clientWidth, img.naturalWidth / 3);
         // If the image is in a column, it might get bigger on smaller screens.
         // We use col-lg for this in snippets, so they get bigger on the md breakpoint
         if (this.$target.closest('[class*="col-lg"]').length) {
```

Comment:
@younn-o Whatever we end up with for the actual resizing of the image, I think you can fix-improve that whole method:
1. The container sizes could be read via CSS variables instead of hardcoded
2. The code does not handle the case where an image is added in a container without columns if I understand correctly (which is possible via our snippets and editor). So in that case, currently, we simply use `img.clientWidth` which is just wrong.
3.  At some point we had a `* 1.5` done on that displayWidth value in that non column / non container parent context to add some tolerance on the width we find, we should restore it we think